kvserver: fix ChangeReplicasTrigger logging

aayushshah15 opened the pull request:
Previously, we would incorrectly log the operation that caused the
change in the range descriptor. This commit fixes this bug.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56930)
<!-- Reviewable:end -->

andreimatei reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56930)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @aayushshah15 and @andreimatei)

___
*[pkg/roachpb/data.go, line 1711 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56930#-MMXHrQ75yphYv65imuN:-MMXHrQ75yphYv65imuO:b-bnk61b) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f24c421727993d64e3382419af26ceb4b0e34d5/pkg/roachpb/data.go#L1711)):*
> ```Go
> 	var nextReplicaID ReplicaID
> 	var afterReplicas []ReplicaDescriptor
> 	added, removed := crt.Added(), crt.Removed()
> ```

can we make these return `ReplicaDescriptors` ?
___
*[pkg/roachpb/data.go, line 1715 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56930#-MMXInSe1ZYxssfTDTHm:-MMXInSe1ZYxssfTDTHn:bvyqia5) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f24c421727993d64e3382419af26ceb4b0e34d5/pkg/roachpb/data.go#L1715)):*
> ```Go
> 	// Voters and non voters are never added/removed in tandem and they're only
> 	// ever added/removed/swapped one at a time.
> 	if len(ReplicaDescriptors{added}.NonVoters()) > 0 {
> ```

This logic is making assumptions that I don't think a stringer for a proto should make about higher-level operations.
I think we should either un-deprecate `DeprecatedChangeType` (or use another `message` or `context` specified when the `ChangeReplicasTrigger` is created) or we don't try to log this higher-level operation. It's not like logging it is very important since it is deduced mechanically from the other things we log.
___
*[pkg/roachpb/data.go, line 1719 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56930#-MMXJvCA4J0RXjQ-w8Va:-MMXJvCA4J0RXjQ-w8Vb:b-mk4r3i) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f24c421727993d64e3382419af26ceb4b0e34d5/pkg/roachpb/data.go#L1719)):*
> ```Go
> 	}
> 	if len(ReplicaDescriptors{removed}.NonVoters()) > 0 {
> 		removeOp = ADD_NON_VOTER
> ```

s/ADD/REMOVE ?


<!-- Sent from Reviewable.io -->

aayushshah15 committed:
kvserver: fix ChangeReplicasTrigger logging

Previously, we would incorrectly log the operation that caused the
change in the range descriptor. This commit fixes this bug by removing
references to the operation in this logging.

Release note: None

aayushshah15 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56930)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei)

___
*[pkg/roachpb/data.go, line 1711 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56930#-MMXHrQ75yphYv65imuN:-MMXODfpEcbWQFFLo6Ee:br9ijaw) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f24c421727993d64e3382419af26ceb4b0e34d5/pkg/roachpb/data.go#L1711)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

can we make these return `ReplicaDescriptors` ?
</blockquote></details>

Not making this change since I'm just removing reference to the operation from the logging (unless you still want me to do it).
___
*[pkg/roachpb/data.go, line 1715 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56930#-MMXInSe1ZYxssfTDTHm:-MMXOAhrDhRrGzsNlJov:b4ef4li) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f24c421727993d64e3382419af26ceb4b0e34d5/pkg/roachpb/data.go#L1715)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

This logic is making assumptions that I don't think a stringer for a proto should make about higher-level operations.
I think we should either un-deprecate `DeprecatedChangeType` (or use another `message` or `context` specified when the `ChangeReplicasTrigger` is created) or we don't try to log this higher-level operation. It's not like logging it is very important since it is deduced mechanically from the other things we log.
</blockquote></details>

makes sense, I removed references to the operation from the logging.
___
*[pkg/roachpb/data.go, line 1719 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56930#-MMXJvCA4J0RXjQ-w8Va:-MMXKRQk9gzoA_sDWhTE:bmswl6e) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f24c421727993d64e3382419af26ceb4b0e34d5/pkg/roachpb/data.go#L1719)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

s/ADD/REMOVE ?
</blockquote></details>

:facepalm: but N/A after addressing previous comments.


<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56930#-:-MMadB5QENDFYM628tzp:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

aayushshah15 commented:
TFTR!

bors r+
